src/main/flight/autopilot.c

```diff
@@ -22,38 +22,141 @@
 
 #include "platform.h"
 #include "build/debug.h"
+#include "common/axis.h"
+#include "common/filter.h"
 #include "common/maths.h"
+#include "common/vector.h"
+#include "fc/rc.h"
+#include "fc/runtime_config.h"
 
 #include "flight/imu.h"
-#include "flight/pid.h"
 #include "flight/position.h"
 #include "rx/rx.h"
+#include "sensors/gyro.h"
 
 #include "autopilot.h"
 
 #define ALTITUDE_P_SCALE  0.01f
 #define ALTITUDE_I_SCALE  0.003f
 #define ALTITUDE_D_SCALE  0.01f
 #define ALTITUDE_F_SCALE  0.01f
+#define POSITION_P_SCALE  0.0012f
+#define POSITION_I_SCALE  0.0001f
+#define POSITION_D_SCALE  0.0015f
+#define POSITION_A_SCALE  0.0008f
+#define UPSAMPLING_CUTOFF_HZ 5.0f
 
 static pidCoefficient_t altitudePidCoeffs;
+static pidCoefficient_t positionPidCoeffs;
+
 static float altitudeI = 0.0f;
 static float throttleOut = 0.0f;
 
+typedef struct {
+    bool isStopping;
+    float distance;
+    float previousDistance;
+    float previousVelocity;
+    float integral;
+    float pidSum;
+    pt1Filter_t velocityLpf;
+    pt1Filter_t accelerationLpf;
+} earthFrame_t;
+
+typedef enum {
+    EW = 0,
+    NS
+} axisEF_t;
+
+typedef struct {
+    float gpsDataIntervalS;
+    float gpsDataFreqHz;
+    float sanityCheckDistance;
+    float upsampleCutoff;
+    float pt1Cutoff;
+    float pt1Gain;
+    bool sticksActive;
+    float maxAngle;
+    float pidSumCraft[EF_AXIS_COUNT];
+    pt3Filter_t upsample[EF_AXIS_COUNT];
+    earthFrame_t efAxis[EF_AXIS_COUNT];
+} posHoldState;
+
+static posHoldState posHold = {
```

Comment:
I'd rather see `posHold.pidSum` than `state.pidSum` since the name `posHold` refers then to all entries in a struct that deal exclusively with posHold.
I was thinking that later we may have other values for other purposes which may be in different structs, so this one is intended for the posHold stuff only.